py4DSTEM/process/phase/parameter_optimize.py

```diff
@@ -0,0 +1,449 @@
+from tqdm import tqdm
+from skopt import gp_minimize
+from skopt.space import Real, Integer, Categorical
+from skopt.plots import (
+    plot_convergence,
+    plot_gaussian_process,
+    plot_evaluations,
+    plot_objective,
+)
+from skopt.utils import use_named_args
+import matplotlib.pyplot as plt
+import numpy as np
+
+from functools import partial
+from typing import Union
+
+from py4DSTEM.process.phase.iterative_base_class import PhaseReconstruction
+from py4DSTEM.process.phase.utils import AffineTransform
+
+
+class PtychographyOptimizer:
+    """
+    Optimize ptychographic hyperparameters with Bayesian Optimization of a
+    Gaussian process. Any of the arguments to the ptychographic
+    init-preprocess-reconstruct pipeline can be optimized over (including
+    boolean or other flag options)
+    """
+
+    def __init__(
+        self,
+        reconstruction_type: type[PhaseReconstruction],
```

Comment:
This is correct as written, so long as the Python version is >= 3.9. See [here](https://mypy.readthedocs.io/en/latest/kinds_of_types.html#the-type-of-class-objects). The optimizer takes the _class_ as an argument, not an _instance_ of the class.
> Sometimes you want to talk about class objects that inherit from a given class. This can be spelled as type[C] (or, on Python 3.8 and lower, [typing.Type[C]](https://docs.python.org/3/library/typing.html#typing.Type)) where C is a class. In other words, when C is the name of a class, using C to annotate an argument declares that the argument is an instance of C (or of a subclass of C), but using type[C] as an argument annotation declares that the argument is a class object deriving from C (or C itself).